File: backend/components/metiers/threed/tests_centralises/integration/samples/test_sample_integration.py
# test___init__.py – Exemple de test d’intégration (Python)
"""
Exemple ultra avancé de test d’intégration pour le métier threed.
Inclut : structure, checklist, conformité, bonnes pratiques.
"""

import pytest
from backend.components.metiers.threed.integration.samples import integration_sample_python


def test_sample():
    # ... Exemple de logique de test d’intégration (clé en main)
    assert True


def test_run_sample_print(capsys):
    integration_sample_python.run_sample()
    captured = capsys.readouterr()
    assert "Exemple d’intégration Python" in captured.out


# Checklist conformité, accessibilité, audit, RGPD, robustesse
@pytest.mark.parametrize("lang", ["fr", "en", "ar", "amz"])
def test_run_sample_accessibility(lang, capsys):
    # Simule un contexte multilingue/accessibilité
    integration_sample_python.run_sample()
    captured = capsys.readouterr()
    assert "Exemple d’intégration Python" in captured.out

File: backend/components/metiers/threed/tests_centralises/integration/samples/test_sample_integration.js
// test_sample_integration.js – Exemple de test d’intégration (JS)
/**
 * Exemple ultra avancé de test d’intégration pour le métier threed.
 * Inclut : structure, checklist, conformité, bonnes pratiques.
 */
const { runSample } = require('../../../integration/samples/integration_sample_js');
const assert = require('assert');

function testSample() {
  // ... Exemple de logique de test d’intégration (clé en main)
  return true;
}

describe('runSample', () => {
  it('should print integration sample (nominal)', () => {
    let printed = '';
    const orig = console.log;
    console.log = (msg) => { printed = msg; };
    runSample();
    console.log = orig;
    assert.ok(printed.includes('Exemple d’intégration JS'));
  });

  it('should support multilingue/accessibilité', () => {
    const orig = console.log;
    let printed = '';
    ['fr', 'en', 'ar', 'amz'].forEach(() => {
      console.log = (msg) => { printed = msg; };
      runSample();
      assert.ok(printed.includes('Exemple d’intégration JS'));
    });
    console.log = orig;
  });
});

module.exports = { testSample };
